### DISCUSSION TITLE
Custom view React Native: How to automated

### DISCUSSION THREAD

**Discussion by samanthamyop** [wix/Detox#3404] [Q&A]
Hi there!
I am coming from a cypress automation background but now the company wants me to automate the web apps as well, so sorry if this is a super noob question. The app has a rule that the user must be 18 or older to register a new user.

The devs are using the date of birth scroller on both the iOS and Android version of the app: 
![Screenshot](https://user-images.githubusercontent.com/102110407/168945785-ec5e03a7-5127-4b8f-9294-b845acbcdde0.png)

The code looks like this:
![code](https://user-images.githubusercontent.com/102110407/168945857-a25f8055-8879-447a-ab35-40ce9298d306.png)

Issue 1:
I have attempted to add testIDs to the individual ScrollablePicker's but when I add the selector into the detox code, I get a no elements match error:

await expect(element(by.id("year"))).toBeVisible();

Issue 2:
Because it's a custom implementation, how would I scroll to say 2000?

Again sorry for the noob question.

**Comment by adomanich** [comment]
@samanthamyop Hello, you can try to select a year in the following way:  

`await element(by.type('UIPickerView')).atIndex(2).setColumnToValue(0, String(year));`
